src/Microsoft.AspNet.Mvc.Core/Internal/ResponseContentTypeHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Internal
+{
+    public static class ResponseContentTypeHelper
+    {
+        public static void ResolveContentTypeAndEncoding(
```

Comment:
I'm not asking for a doc comment here, but a little explanation of what the purpose of this is, and what scenarios it handles.